### PR TITLE
Feat: Update hex strings to hex enncoded string type

### LINKED ISSUE #944: Add a HexString type for strings that are hex encoded
## Description

Add a HexString type for strings that are hex encoded

## Motivation

Makes it clearer to work with

## Requirements

1. Fields that are currently a `string`, but are hex encoded with `0x` prefix, should get a new HexString type (or similar name)

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by abbasabid**
# Description of change

- Updated the strings that are hex encoded with `0x` prefix to a new `HexEncodedString` type.
- Also updated places that use HexEncodedAmount.

fixes https://github.com/iotaledger/iota.js/issues/944

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

### CODE REVIEW

File: packages/iota/src/models/payloads/ITaggedDataPayload.ts [hunk 1]
```diff
@@ -14,7 +15,7 @@ export interface ITaggedDataPayload extends ITypeBase<5> {
     /**
      * The tag to use to categorize the data.
      */
-    tag: string;
+    tag: HexEncodedString;
 
     /**
      * The index data.
```
Comment (Thoralf-M) [inline packages/iota/src/models/payloads/ITaggedDataPayload.ts:21]: Shouldn't this also be a HexEncodedString?

Comment (abbasabid) [reply packages/iota/src/models/payloads/ITaggedDataPayload.ts:21]: I think i missed it. Please review the PR and let me know if i miss any other places as well. Thanks.

File: packages/iota/src/models/payloads/IMilestonePayload.ts [hunk 2]
```diff
@@ -31,22 +32,22 @@ export interface IMilestonePayload extends ITypeBase<7> {
...
+    inclusionMerkleRoot: HexEncodedString;
 
     /**
      * The Merkle tree hash of all blocks applied by this milestone.
      */
-    appliedMerkleRoot: string;
+    appliedMerkleRoot: HexEncodedString;
 
     /**
      * The metadata.
```
Comment (Thoralf-M) [inline packages/iota/src/models/payloads/IMilestonePayload.ts:53]: This should also be an HexEncodedString

Comment (Thoralf-M) [inline packages/iota/src/models/payloads/IMilestonePayload.ts:53]: This should also be an HexEncodedString

File: packages/iota/src/models/outputs/IAliasOutput.ts [hunk 3]
```diff
@@ -21,7 +22,7 @@ export interface IAliasOutput extends ITypeBase<4>, ICommonOutput {
     /**
      * Unique identifier of the alias, which is the BLAKE2b-160 hash of the Output ID that created it.
      */
-    aliasId: string;
+    aliasId: HexEncodedString;
 
     /**
      * A counter that must increase by 1 every time the alias is state transitioned.
```
Comment (Thoralf-M) [inline packages/iota/src/models/outputs/IAliasOutput.ts:28]: Can't comment there, but `stateMetadata` should also be updated

File: packages/iota/src/models/payloads/ITaggedDataPayload.ts (hunk 1 above)
Comment (abbasabid) [reply packages/iota/src/models/payloads/ITaggedDataPayload.ts:21]: Fixed.

File: packages/iota/src/models/payloads/IMilestonePayload.ts (hunk 2 above)
Comment (abbasabid) [reply packages/iota/src/models/payloads/IMilestonePayload.ts:53]: Fixed.

Comment (abbasabid) [reply packages/iota/src/models/payloads/IMilestonePayload.ts:53]: Fixed.

File: packages/iota/src/models/outputs/IAliasOutput.ts (hunk 3 above)
Comment (abbasabid) [reply packages/iota/src/models/outputs/IAliasOutput.ts:28]: Fixed.

File: packages/iota/src/models/tokenSchemes/ISimpleTokenScheme.ts [hunk 4]
```diff
@@ -14,15 +15,15 @@ export interface ISimpleTokenScheme extends ITypeBase<0> {
     /**
      * Amount of tokens minted by this foundry.
      */
-    mintedTokens: string;
+    mintedTokens: HexEncodedString;
```
Comment (Thoralf-M) [inline packages/iota/src/models/tokenSchemes/ISimpleTokenScheme.ts]: For the amount we have `HexEncodedAmount`, same for the fields below

File: packages/iota/src/models/hexEncodedString.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * Hex encoded U256.
```
Comment (Thoralf-M) [inline packages/iota/src/models/hexEncodedString.ts]:
```suggestion
 * Hex encoded bytes.
```

File: packages/iota/src/models/hexEncodedTypes.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * Hex encoded bytes.
+ */
+export type HexEncodedString = string;
+
+/**
+ * Hex encoded bytes.
```
Comment (Thoralf-M) [inline packages/iota/src/models/hexEncodedTypes.ts]:
```suggestion
 * Hex encoded U256.
```

**Review by Thoralf-M — APPROVED** [review]

**Review by msarcev — APPROVED** [review]

**Pull request merged**